src/webgpu/api/operation/labels.spec.ts

```diff
@@ -3,10 +3,271 @@ Tests for object labels.
+         return vec4f(0, 0, 0, 1);
+        }
+      `,
+    });
+    t.expect(shaderModule.label === label);
+  },
+
+  createComputePipeline: async (t: GPUTest, label: string) => {
+    const module = t.device.createShaderModule({
+      code: `
+        @compute @workgroup_size(1u) fn foo() {}
+      `,
+    });
+    const computePipeline = t.device.createComputePipeline({
+      label,
+      layout: 'auto',
+      compute: {
+        module,
+        entryPoint: 'foo',
+      },
+    });
+    t.expect(computePipeline.label === label);
+  },
+
+  createRenderPipeline: async (t: GPUTest, label: string) => {
+    const module = t.device.createShaderModule({
+      code: `
+        @vertex fn foo() -> @builtin(position) vec4f {
+         return vec4f(0, 0, 0, 1);
+        }
+      `,
+    });
+    const renderPipeline = t.device.createRenderPipeline({
+      label,
+      layout: 'auto',
+      vertex: {
+        module,
+        entryPoint: 'foo',
+      },
+    });
+    t.expect(renderPipeline.label === label);
+  },
+
+  createComputePipelineAsync: async (t: GPUTest, label: string) => {
+    const module = t.device.createShaderModule({
+      code: `
+        @compute @workgroup_size(1u) fn foo() {}
+      `,
+    });
+    const computePipeline = await t.device.createComputePipelineAsync({
+      label,
+      layout: 'auto',
+      compute: {
+        module,
+        entryPoint: 'foo',
+      },
+    });
+    t.expect(computePipeline.label === label);
+  },
+
+  createRenderPipelineAsync: async (t: GPUTest, label: string) => {
+    const module = t.device.createShaderModule({
+      label,
+      code: `
+        @vertex fn foo() -> @builtin(position) vec4f {
+         return vec4f(0, 0, 0, 1);
+        }
+      `,
+    });
+    const renderPipeline = await t.device.createRenderPipelineAsync({
+      label,
+      layout: 'auto',
+      vertex: {
+        module,
+        entryPoint: 'foo',
+      },
+    });
+    t.expect(renderPipeline.label === label);
+  },
+
+  createCommandEncoder: async (t: GPUTest, label: string) => {
+    const encoder = t.device.createCommandEncoder({ label });
+    t.expect(encoder.label === label);
+  },
+
+  createRenderBundleEncoder: async (t: GPUTest, label: string) => {
+    const encoder = t.device.createRenderBundleEncoder({
+      label,
+      colorFormats: ['rgba8unorm'],
+    });
+    t.expect(encoder.label === label);
+  },
+
+  createQuerySet: async (t: GPUTest, label: string) => {
+    const querySet = t.device.createQuerySet({
+      label,
+      type: 'occlusion',
+      count: 1,
+    });
+    t.expect(querySet.label === label);
+  },
+
+  beginRenderPass: async (t: GPUTest, label: string) => {
+    const encoder = t.device.createCommandEncoder();
+    const renderPass = encoder.beginRenderPass({
+      label,
+      colorAttachments: [],
+    });
+    t.expect(renderPass.label === label);
+    renderPass.end();
+    t.expect(renderPass.label === label);
```

Comment:
check again after ending the encoder as well? same for compute pass

Reply:
done

Reply:
Sorry what I meant was like,
```
    t.expect(renderPass.label === label);
    renderPass.end();
    t.expect(renderPass.label === label);
    encoder.finish();
    t.expect(renderPass.label === label);
```

though this is probably useful too

Reply:
oh. haha 😅

Reply:
done